### PR TITLE
[FIX] loyalty: Prevent reward_ids to be reset to default

### PR DESCRIPTION
**Pull request by abz-odoo**
Steps:
    - Install `sale_management`
    - Go to `Discount & Loyalty`
    - Create a new one and set a name
    - Change default program type from `discount` to `buy x get y`
    - Remove the default rewards
    - Add a new one (the default one)
    - Click `Save & Close`
    - Trigger save

Rewards is reset to default `discount` instead of `buy x get y`

### CODE REVIEW

**Comment by abz-odoo** [comment]
@odoo/rd-sales

**Review by vava-odoo — APPROVED** [review]
Hi @abz-odoo 
Thanks for this. Looks very similar to [this bugfix PR](https://github.com/odoo/odoo/pull/116244).
That's the kind of things we really need to cover with a test. Could you adapt `test_loyalty_program_preserve_reward_upon_writing` or add a similar one?
Cheers

**Comment by abz-odoo** [comment]
Hello @vava-odoo, thanks for reviewing this pr.

Test is done

File: addons/loyalty/tests/test_loyalty.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 from odoo.fields import Command
 from odoo.tests import tagged, TransactionCase
 from odoo.tools import mute_logger
-
+from odoo.tests.common import Form
```
Comment (vava-odoo) [inline addons/loyalty/tests/test_loyalty.py:10]: Can be imported from odoo.tests directly, at line 8

**Review by vava-odoo — APPROVED** [review]
Thanks @abz-odoo!

**Comment by vava-odoo** [comment]
@abz-odoo Looks like the runbot is very red :open_mouth:

File: addons/loyalty/models/loyalty_program.py [hunk 2]
```diff
@@ -400,6 +400,18 @@ def toggle_active(self):
             program.reward_ids.with_context(active_test=True).discount_line_product_id.active = program.active
         return res
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if (
+                len(vals_list) == 1 and 'reward_ids' in vals and 'program_type' in vals
+                and self._fields['reward_ids'].convert_to_cache(vals['reward_ids'], self)
+            ):
```
Comment (Feyensv) [inline addons/loyalty/models/loyalty_program.py:409]: Don't loop on vals_list if its length is 1 ;)

**Review by Feyensv — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**